docs/tooling-consolidation.md

```diff
@@ -0,0 +1,82 @@
+# AspNetCore Tooling Consolidation
+
+## Objectives
+
+We want to consolidate portions of dotnet/aspnetcore-tooling into dotnet/aspnetcore in our ongoing effort to reduce overall build time end to end for the .NET Core SDK. As such, we will not require all of aspnetcore-tooling to be merged into aspnetcore. For example, tooling for VSMac and VSCode will remain in aspnetcore-tooling. The main motivation is to achieve the overall goals of repo consolidation to reduce the number of repository required to build the SDK while keeping the build in aspnetcore as simple as possible without needing to support build/test for tooling scenarios such as testing on VSCode. There is also an added benefit of maintaining the current development and release workflow for aspnetcore-tooling which is more "agile" than aspnetcore (e.g. faster PR builds, release cycles that synchronize with VS releases).
+
+To achieve this we will be migrating the following (and associated tests) from aspnetcore-tooling to aspnetcore:
+```
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions
+Microsoft.AspNetCore.Razor.Language
+Microsoft.AspNetCore.Razor.Tools
+Microsoft.CodeAnalysis.Razor
+Microsoft.NET.Sdk.Razor
+````
+The following (and associated tests) will remain in aspnetcore-tooling:
+```
+Microsoft.AspNetCore.Razor.LanguageServer.Common
+Microsoft.AspNetCore.Razor.LanguageServer
+Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed
+Microsoft.AspNetCore.Razor.OmniSharpPlugin
+Microsoft.AspNetCore.Razor.VSCode.Extension
+Microsoft.AspNetCore.Razor.VSCode
+Microsoft.CodeAnalysis.Razor.Workspaces
+Microsoft.CodeAnalysis.Remote.Razor
+Microsoft.VisualStudio.Editor.Razor
+Microsoft.VisualStudio.LanguageServerClient.Razor
+Microsoft.VisualStudio.LanguageServices.Razor
+Microsoft.VisualStudio.LiveShare.Razor
+Microsoft.VisualStudio.Mac.LanguageServices.Razor
+Microsoft.VisualStudio.Mac.RazorAddin
+Microsoft.VisualStudio.RazorExtension
+RazorDeveloperTools
+rzls
+````
```

Comment:
How was the above split determined exactly?

Reply:
Whether we need any of this in building aspnetcore. @NTaylorMullen took a look at all the projects and determined that thees need to be migrated according to that criterion.